base/workerpool.jl

```diff
@@ -0,0 +1,85 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+type WorkerPool
+    channel::RemoteChannel{Channel{Int}}
+end
+
+
+isbusy(pool::WorkerPool) = !isready(pool.channel)
+
+
+"""
+    WorkerPool(workers)
+
+Create a WorkerPool from a vector of worker ids.
+"""
+function WorkerPool(workers::Vector{Int})
+
+    # Create a shared queue of workers...
+    pool = RemoteChannel(()->Channel{Int}(128)) 
+
+    # Check that workers are not already part of a pool...
+    check = () -> if :_worker_pool in names(Main)
+        error("Worker $(myid()) already in a WorkerPool!")
+    end
+    foreach(fetch, [@spawnat w check() for w in workers])
+
+    # Put each worker into the pool...
+    for w in workers
+        put!(pool, w)
+        @spawnat w global _worker_pool = pool
+    end
+
+    WorkerPool(pool)
+end
+
+
+"""
+    WorkerPool(n)
+
+Launches `n` workers using the in-built LocalManager.
+"""
+WorkerPool(n::Integer) = WorkerPool(addprocs(n))
+
+
+"""
```

Comment:
The intention here was just to mirror the existing `addprocs(n)` behaviour.
Happy to drop it.